### ISSUE TITLE
bug: rules configuration is ignored in nested directories (e.g., in test/)

### ISSUE THREAD

**Issue by solid-illiaaihistov** [solid-software/solid_lints#309]
File Structure
```text
my_project/
  ├── analysis_options.yaml
  ├── lib/
  └── test/
       ├── analysis_options.yaml
       └── my_test.dart
```
analysis_options.yaml (Root)

```yaml
analyzer:
  plugins:
    - solid_lints
solid_lints:
  diagnostics:
    avoid_late_keyword: true
```

test/analysis_options.yaml (Nested)
```yaml
solid_lints:
  diagnostics:
    avoid_late_keyword: false
```

test/my_test.dart
```dart
void main() {
  late String myMock;
}
```

### Expected behavior
No lint is triggered for test/my_test.dart because avoid_late_keyword is disabled in the nested test/analysis_options.yaml.

### Actual behavior
The avoid_late_keyword lint rule is triggered for test/my_test.dart because the nested analysis_options.yaml is completely ignored, and the rules are evaluated using the root configuration.
